### ISSUE TITLE
Textsize() with custom fonts

### ISSUE THREAD

**Issue by Alivanar** [processing/p5.js#2055]
#### Nature of issue?

- [x] Found a bug
- [ ] Existing feature enhancement
- [ ] New feature request

#### Most appropriate sub-area of p5.js?

- [ ] Color
- [ ] Core
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [x] Typography
- [ ] Utilities
- [ ] WebGL

#### Which platform were you using when you encountered this?

- [ ] Mobile/Tablet (touch devices)
- [x] Desktop/Laptop
- [ ] Others (specify if possible)

#### Details about the bug: 

- p5.js version: p5.js v0.5.11 June 01, 2017
- Web browser and it's version: Vivaldi 1.10.867.46 (Stable channel) (64-bit) (Will test it on firefox soon)
- Operating System: Elementary OS (Linux)
- Steps to reproduce this: 

sketch.js :

    captureItReady = false;
    
    function preload(){
      captureIt = loadFont("./fonts/Capture_it.ttf", captureItRead);
    };
    
    function setup() {
      createCanvas(800,500);
      textAlign(CENTER, CENTER);
    };
    
    function draw() {
     background(51);
     textSize(12);
     if(captureItReady){
       textFont(captureIt);
       textSize(30);
       fill(255);
       text("TITLE", width/2, height/2);
     }else{
       text("Failed to load font", width/2, height/2);
     };
     textFont('Georgia');
     textSize(50);
     text("Blah", width/2, 2*height/3);
    };
    
    function captureItRead(){
      captureItReady = true;
    };

Expected Behavior : First piece of text will be sized 30 and second piece of text will be sized 50
Actual Behavior : First piece is 30, but second is 12 (textSize called at the beginning of the code)

Here is the link to my post on the processing forum where i tested some other variations of this code, with more or less the same behavior at the end : 
[https://forum.processing.org/two/discussion/comment/101962#Comment_101962](url)

I think it has to do with custom fonts (loaded from external .otf or .ttf files), but maybe you'll be able to tell more about it ?

**Comment by dhowe** [comment]
This is a bit of a strange test as there is no need for a callback in preload(), as preload() is guaranteed to finish before setup() is called. Nonetheless, I can verify the problem. 

Here is a simpler test:
```
var font;

function preload() {
  font = loadFont("anyfont.ttf");
}

function setup() {
  createCanvas(400, 400);

  textFont(font);
  textSize(30);
  text("Foo", 180, 200);

  textFont('Georgia');
  textSize(50);
  text("Bar", 180, 250);
}
```

The problem only occurs when the first font is loaded/opentype and the second is a browser font.

**Comment by seymar** [comment]
I also encountered this problem.

I fixed it by using CSS `@font-face` to add the font to the page instead of using the `loadFont` function.

**Comment by dhowe** [comment]
This should be fixed in branch dhowe-2055 
@seymar @Alivanar please verify...

**Comment by lmccart** [comment]
confirming this is fixed with #2055
